pkg/controller/restore_controller.go

```diff
@@ -330,42 +349,62 @@ func (controller *restoreController) fetchBackup(bucket, name string) (*api.Back
 }
 
 func (controller *restoreController) runRestore(restore *api.Restore, bucket string) (warnings, errors api.RestoreResult) {
+	logContext := controller.logger.WithField("restore", kubeutil.NamespaceAndName(restore))
+
 	backup, err := controller.fetchBackup(bucket, restore.Spec.BackupName)
 	if err != nil {
-		glog.Errorf("error getting backup: %v", err)
+		logContext.WithFields(logrus.Fields{
+			"backup": restore.Spec.BackupName,
+			"error":  err,
+		}).Error("Error getting backup")
 		errors.Ark = append(errors.Ark, err.Error())
 		return
 	}
 
-	tmpFile, err := downloadToTempFile(restore.Spec.BackupName, controller.backupService, bucket)
+	tmpFile, err := downloadToTempFile(restore.Spec.BackupName, controller.backupService, bucket, controller.logger)
 	if err != nil {
-		glog.Errorf("error downloading backup: %v", err)
+		logContext.WithFields(logrus.Fields{
+			"backup": restore.Spec.BackupName,
+			"error":  err,
+		}).Error("Error downloading backup")
 		errors.Ark = append(errors.Ark, err.Error())
 		return
 	}
 
 	logFile, err := ioutil.TempFile("", "")
 	if err != nil {
-		glog.Errorf("error creating log temp file: %v", err)
+		logContext.WithField("error", err).Error("Error creating log temp file")
 		errors.Ark = append(errors.Ark, err.Error())
 		return
 	}
 
 	defer func() {
 		if err := tmpFile.Close(); err != nil {
```

Comment:
tech debt: i think this needs to be rethought. If tmpfile.Close an error then the data was not written to to cleanly and may be corrupted. Same with the rest of this defer block.

Reply:
Do you have suggestions for how to do this better? I'm not sure we can do much other than throw our hands up in the air?